rewrite-yaml/src/main/java/org/openrewrite/yaml/MergeYamlVisitor.java

```diff
@@ -298,46 +308,42 @@ private Yaml.Sequence mergeSequence(Yaml.Sequence s1, Yaml.Sequence s2, P p, Cur
-                    for (Yaml.Sequence.Entry existingEntry : s1.getEntries()) {
-                        Yaml.Mapping existingMapping = (Yaml.Mapping) existingEntry.getBlock();
-                        if (keyMatches(existingMapping, incomingMapping)) {
-                            Yaml.Sequence.Entry e1 = existingEntry.withBlock(mergeMapping(existingMapping, incomingMapping, p, cursor));
-                            if (e1 == existingEntry) {
-                                // Made no change, no need to consider the entry "mutated"
-                                return null;
-                            }
-                            return e1;
-                        }
-                    }
-                    return entry;
-                });
-                if (mutatedEntries.isEmpty()) {
-                    return s1;
-                }
+            List<Yaml.Sequence.Entry> newEntries = map(incomingEntries, it -> autoFormat(it, p, cursor));
+            List<Yaml.Sequence.Entry> mutatedEntries = concatAll(s1.getEntries(), newEntries, it -> ((Yaml.Scalar) it.getBlock()).getValue()).ls;
 
-                List<Yaml.Sequence.Entry> entries = concatAll(
-                        s1.getEntries().stream().filter(entry -> !mutatedEntries.contains(entry)).collect(Collectors.toList()),
-                        map(mutatedEntries, entry -> autoFormat(entry, p, cursor)));
+            return s1.withEntries(mutatedEntries);
+        }
 
-                if (entries.size() != s1.getEntries().size()) {
-                    return s1.withEntries(entries);
-                }
-                for (int i = 0; i < s1.getEntries().size(); i++) {
-                    if (entries.get(i) != s1.getEntries().get(i)) {
-                        return s1.withEntries(entries);
+        if (objectIdentifyingProperty == null) {
+            // No identifier set to match entries on, so cannot continue
+            return s1;
+        }
+
+        List<Yaml.Sequence.Entry> mutatedEntries = map(s2.getEntries(), entry -> {
+            Yaml.Mapping incomingMapping = (Yaml.Mapping) entry.getBlock();
+            for (Yaml.Sequence.Entry existingEntry : s1.getEntries()) {
+                Yaml.Mapping existingMapping = (Yaml.Mapping) existingEntry.getBlock();
+                if (keyMatches(existingMapping, incomingMapping)) {
+                    Yaml.Sequence.Entry e1 = existingEntry.withBlock(mergeMapping(existingMapping, incomingMapping, p, cursor));
+                    if (e1 == existingEntry) {
+                        // Made no change, no need to consider the entry "mutated"
+                        return null;
                     }
+                    return e1;
                 }
-                return s1;
             }
-        }
+            return entry;
+        });
+
+        List<Yaml.Sequence.Entry> entries = concatAll(
+                filter(s1.getEntries(), it -> !mutatedEntries.contains(it)),
```

Comment:
Do we really need a new explicit `ListUtils.filter` operation used only once if we could also use
```suggestion
                map(s1.getEntries(), it -> mutatedEntries.contains(it) ? null : it),
```
I'm looking to minimize the API surface we're adding and maintaining going forward. A bit on the fence on this one.

Reply:
I didn't even consider doing this 🧐. Technically you are right, but it really feels like a hack. But I can change it of course, if you are adement about this one!

---

Sidenote: That's my problem with the ListUtils right now, the contract of the functions are not confirm to the mathematical laws. A `map` actions to a list of x items should always return a [new] list with again x items. It's a little bit of nitpicking of course, but it does not make the code readable if you start adding these kind of side-effects to universal concepts like mapping/flat-mapping.

Reply:
Mapping elements to `null` is quite common throughout our code base already, but perhaps indeed it helps to be explicit. Hence why I was a bit on the fence here. Ok to add this then I suppose and gradually adopt.